assets/js/theme/global/adminBar.js

```diff
@@ -0,0 +1,103 @@
+/**
+ * Persistent admin bar for store administrators viewing the storefront
+ * @param maintenanceMode
+ * @param secureBaseUrl
+ * @param adminBarLanguage
+ * @param productId
+ * @param categoryId
+ */
+export default function (secureBaseUrl, maintenanceMode = {}, adminBarLanguage, productId, categoryId) {
+    const lang = adminBarLanguage.translations;
+    const password = maintenanceMode.password || false;
+    const isMaintenanceMode = (maintenanceMode.header && !password) || false;
+
+    function isInIframe() {
+        try {
+            return window.location !== window.parent.location;
+        } catch (e) {
+            return true;
+        }
+    }
+
+    // eslint-disable-next-line no-shadow
+    function getContextualMatch(productId, categoryId) {
+        if (productId && parseInt(productId, 10)) {
+            return {
+                type: 'product',
+                url: `/manage/products/${productId}/edit`,
+            };
+        } else if (categoryId && parseInt(categoryId, 10)) {
+            return {
+                type: 'category',
+                url: `/manage/products/categories/${categoryId}/edit`,
+            };
+        }
+    }
+
+    function doesNotHaveAdminCookie() {
+        return document.cookie.indexOf('ADMIN_BAR=1') === -1;
+    }
+
+    // Set admin cookie if appropriate
+    if (window.URLSearchParams && window.URL && (new URL(document.location)).searchParams.get('ctk')) {
```

Comment:
I'm simply matching on the query param used when we launch the storefront from the CP. While this would be easy to forge, it doesn't really reveal anything particularly sensitive - every button you can click loads the CP, which is behind the merchant login.